static/app/components/events/interfaces/breadcrumbs/breadcrumb/data/default.tsx

```diff
@@ -48,50 +49,80 @@ function isEventId(maybeEventId: string): boolean {
   return /^[a-fA-F0-9]{32}$/.test(maybeEventId);
 }
 
-const FormatMessage = withProjects(function FormatMessageInner({
+type DataItem = {
+  id: string;
+  'project.name': string;
+  title: string;
+};
+
+function FormatMessage({
   searchTerm,
   event,
   message,
   breadcrumb,
-  projects,
-  loadingProjects,
   orgSlug,
 }: {
   breadcrumb: BreadcrumbTypeDefault | BreadcrumbTypeNavigation;
-  loadingProjects: boolean;
   message: string;
   orgSlug: Organization['slug'];
-  projects: Project[];
   searchTerm: string;
   event?: Event;
 }) {
+  const {projects, fetching: loadingProjects} = useProjects();
+
   const content = <Highlight text={searchTerm}>{message}</Highlight>;
-  if (
-    !loadingProjects &&
-    breadcrumb.category === 'sentry.transaction' &&
-    isEventId(message)
-  ) {
-    const maybeProject = projects.find(project => {
-      return event && project.id === event.projectID;
-    });
 
+  const isSentryTransaction =
+    breadcrumb.category === 'sentry.transaction' && isEventId(message);
+
+  const maybeProject = !loadingProjects
+    ? projects.find(project => {
+        return event && project.id === event.projectID;
+      })
+    : null;
+
+  const {data: queryData} = useApiQuery<{data: DataItem[]; meta: any}>(
```

Comment:
One thing I'm worried about here is if there are events with a large number of unique transaction IDs. If someone were to scroll down the list quickly, it will cause a large number of requests to go off at once. If the number of unique transactions is <5 it doesn't really matter, but if you could have 10+ it might be an issue.

The _best_ way to do this would be to implement batching (make a request for all visible transaction IDs with some sort of debouncing), but that would complicate the logic quite a bit. And I'm not sure how we would want to do that with React Query (as an aside, this could be a good feature addition).

The other option would be to lift this query up to the `Breadcrumbs` component and do a single request for all transaction IDs at once. There's a chance that a query like that would not be that much slower than each individual query so it could be worth a shot.